From 893300abc530276c03fadcc1c8c3a5e6d5ca186c Mon Sep 17 00:00:00 2001
From: Richard Hansen <rhansen@bbn.com>
Date: Mon, 29 Jun 2015 13:07:55 -0400
Subject: [PATCH] eliminiate excess leading blank lines from indented lists

Also, add a test for indented lists.  The HTML output is all manner of
broken, but it is already horribly broken for other lists.
---
 tests/input/list_format.xml     | 14 ++++++++++++++
 tests/valid/list_format.html    | 14 +++++++++++++-
 tests/valid/list_format.nroff   | 15 ++++++++++++++-
 tests/valid/list_format.raw.txt | 11 ++++++++++-
 tests/valid/list_format.txt     | 11 ++++++++++-
 xml2rfc/writers/raw_txt.py      |  4 ++++
 6 files changed, 65 insertions(+), 4 deletions(-)

diff --git a/tests/input/list_format.xml b/tests/input/list_format.xml
index 014dde0..5db82de 100644
--- a/tests/input/list_format.xml
+++ b/tests/input/list_format.xml
@@ -86,6 +86,20 @@ Different list formatting characters, following what v1 supports:
       </t>
    </list>
 
+<vspace blankLines="2"/>
+
+
+   indented list:
+   <list style="empty">
+     <t>
+       <list style="symbols">
+         <t>item one</t>
+         <t>item two</t>
+       </list>
+     </t>
+   </list>
+   end of indented list
+
 
 </t>
 
diff --git a/tests/valid/list_format.html b/tests/valid/list_format.html
index 9528a6f..9c0d5bd 100644
--- a/tests/valid/list_format.html
+++ b/tests/valid/list_format.html
@@ -1,4 +1,4 @@
-<p>First list: <br/><br/> Different list formatting characters, following what v1 supports: <br/><br/><br/> style="format %i": <br/><br/><br/> style="format (%d)": <br/><br/><br/> style="format %c)": <br/><br/><br/> style="format %C.": <br/><br/><br/> style="format %d.": </p>
+<p>First list: <br/><br/> Different list formatting characters, following what v1 supports: <br/><br/><br/> style="format %i": <br/><br/><br/> style="format (%d)": <br/><br/><br/> style="format %c)": <br/><br/><br/> style="format %C.": <br/><br/><br/> style="format %d.": <br/><br/><br/> indented list: </p>
 
 <dl>
   <dt>R1   </dt>
@@ -87,3 +87,15 @@
 </dl>
 
 <p> </p>
+
+<ul class="empty">
+  <li>
+    <ul>
+      <li>item one</li>
+      <li>item two</li>
+    </ul>
+    <p> </p>
+  </li>
+</ul>
+
+<p> end of indented list </p>
diff --git a/tests/valid/list_format.nroff b/tests/valid/list_format.nroff
index 165a3b8..7a2ff68 100644
--- a/tests/valid/list_format.nroff
+++ b/tests/valid/list_format.nroff
@@ -124,4 +124,17 @@ A.  first \%sub-item
 B.  second \%sub-item
 .ti 7
 
-C.  third \%sub-item
\ No newline at end of file
+C.  third \%sub-item
+.sp 2
+.in 3
+indented list:
+.in 9
+.ti 6
+
+*  item one
+.ti 6
+
+*  item two
+.in 3
+
+end of indented list
\ No newline at end of file
diff --git a/tests/valid/list_format.raw.txt b/tests/valid/list_format.raw.txt
index 4570a8f..176ab88 100644
--- a/tests/valid/list_format.raw.txt
+++ b/tests/valid/list_format.raw.txt
@@ -83,4 +83,13 @@
 
        B.  second sub-item
 
-       C.  third sub-item
\ No newline at end of file
+       C.  third sub-item
+
+
+   indented list:
+
+      *  item one
+
+      *  item two
+
+   end of indented list
\ No newline at end of file
diff --git a/tests/valid/list_format.txt b/tests/valid/list_format.txt
index 4570a8f..176ab88 100644
--- a/tests/valid/list_format.txt
+++ b/tests/valid/list_format.txt
@@ -83,4 +83,13 @@
 
        B.  second sub-item
 
-       C.  third sub-item
\ No newline at end of file
+       C.  third sub-item
+
+
+   indented list:
+
+      *  item one
+
+      *  item two
+
+   end of indented list
\ No newline at end of file
diff --git a/xml2rfc/writers/raw_txt.py b/xml2rfc/writers/raw_txt.py
index 1021db3..bb3253d 100644
--- a/xml2rfc/writers/raw_txt.py
+++ b/xml2rfc/writers/raw_txt.py
@@ -184,6 +184,10 @@ class RawTextRfcWriter(BaseRfcWriter):
             elif counter_index not in self.list_counters:
                 # Initialize if we need to
                 self.list_counters[counter_index] = 0
+        elif style == 'empty':
+            # empty is only used for indentation
+            indent += 3
+            bullet = ''
         t_count = 0
         for element in list:
             # Check for PI
-- 
2.4.5

